tests/components/airthings_ble/test_sensor.py

```diff
@@ -0,0 +1,213 @@
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+    assert len(hass.states.async_all()) > 0
+
+    assert (
+        entity_registry.async_get(sensor.entity_id).unique_id
+        == WAVE_DEVICE_INFO.address + "_temperature"
+    )
+
+
+async def test_migration_from_v2_to_v3_unique_id(hass: HomeAssistant):
+    """Verify that we can migrate from v2 (introduced in 2023.9.0) to the latest unique id format."""
+    entry = create_entry(hass)
+    device = create_device(hass, entry)
+
+    assert entry is not None
+    assert device is not None
+
+    entity_registry = hass.helpers.entity_registry.async_get(hass)
+
+    await hass.async_block_till_done()
+
+    sensor = entity_registry.async_get_or_create(
+        domain=DOMAIN,
+        platform="sensor",
+        unique_id=HUMIDITY_V2.unique_id,
+        config_entry=entry,
+        device_id=device.id,
+    )
+
+    await hass.async_block_till_done()
+    assert len(hass.states.async_all()) == 0
+
+    inject_bluetooth_service_info(
+        hass,
+        WAVE_SERVICE_INFO,
+    )
+
+    await hass.async_block_till_done()
+
+    with patch_airthings_device_update():
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+    assert len(hass.states.async_all()) > 0
+
+    assert (
+        entity_registry.async_get(sensor.entity_id).unique_id
+        == WAVE_DEVICE_INFO.address + "_humidity"
+    )
+
+
+async def test_migration_from_v1_and_v2_to_v3_unique_id(hass: HomeAssistant):
+    """Test if migration works when we have both v1 (pre 2023.9.0) and v2 (introduced in 2023.9.0) unique ids."""
+    entry = create_entry(hass)
+    device = create_device(hass, entry)
+
+    assert entry is not None
+    assert device is not None
+
+    entity_registry = hass.helpers.entity_registry.async_get(hass)
+
+    await hass.async_block_till_done()
+
+    v2 = entity_registry.async_get_or_create(
+        domain=DOMAIN,
+        platform="sensor",
+        unique_id=CO2_V2.unique_id,
+        config_entry=entry,
+        device_id=device.id,
+    )
+
+    v1 = entity_registry.async_get_or_create(
+        domain=DOMAIN,
+        platform="sensor",
+        unique_id=CO2_V1.unique_id,
+        config_entry=entry,
+        device_id=device.id,
+    )
+
+    await hass.async_block_till_done()
+    assert len(hass.states.async_all()) == 0
+
+    inject_bluetooth_service_info(
+        hass,
+        WAVE_SERVICE_INFO,
+    )
+
+    await hass.async_block_till_done()
+
+    with patch_airthings_device_update():
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+    assert len(hass.states.async_all()) > 0
+
+    assert (
+        entity_registry.async_get(v1.entity_id).unique_id
+        == WAVE_DEVICE_INFO.address + "_co2"
+    )
+    assert entity_registry.async_get(v2.entity_id).unique_id == v2.unique_id
```

Comment:
I'd store the expected unique id in a local variable so we don't rely on the details of the entity registry implementation creating new entity registry entry instances on update. If that would not happen, this check would be useless as it would compare the same attribute on the same instance.